src/library.js

```diff
@@ -494,44 +494,27 @@ LibraryManager.library = {
   },
 #else
 
+  // Grows the asm.js/wasm heap to the given byte size, and updates both JS and asm.js/wasm side views to the buffer.
+  // Returns 1 on success, or undefined if growing failed.
   $emscripten_realloc_buffer: function(size) {
-#if WASM
-    var PAGE_MULTIPLE = {{{ getPageSize() }}};
-    size = alignUp(size, PAGE_MULTIPLE); // round up to wasm page size
-    var oldSize = buffer.byteLength;
-    // native wasm support
-    // note that this is *not* threadsafe. multiple threads can call .grow(), and each
-    // presents a delta, so in theory we may over-allocate here (e.g. if two threads
-    // ask to grow from 256MB to 512MB, we get 2 requests to add +256MB, and may end
-    // up growing to 768MB (even though we may have been able to make do with 512MB).
-    // TODO: consider decreasing the step sizes in emscripten_resize_heap
     try {
-      var result = wasmMemory.grow((size - oldSize) / {{{ WASM_PAGE_SIZE }}}); // .grow() takes a delta compared to the previous size
-      if (result !== (-1 | 0)) {
-        // success in native wasm memory growth, get the buffer from the memory
-        buffer = wasmMemory.buffer;
-        return true;
-      } else {
-        return false;
-      }
-    } catch(e) {
-#if ASSERTIONS
-      console.error('emscripten_realloc_buffer: Attempted to grow from ' + oldSize  + ' bytes to ' + size + ' bytes, but got error: ' + e);
-#endif
-      return false;
-    }
+#if WASM
+      // round size grow request up to wasm page size (fixed 64KB per spec)
+      wasmMemory.grow((size - buffer.byteLength + 65535) >> 16); // .grow() takes a delta compared to the previous size
```

Comment:
Might be nice to use `{{{ WASM_PAGE_SIZE - 1 }}}` instead of `65535`, and `{{{ Math.log2(WASM_PAGE_SIZE) }}}` instead of `16`. But maybe that doesn't help readability, and this value is not expected to ever change.